pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol.go

```diff
@@ -15,11 +15,11 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-func ProvideService(features featuremgmt.FeatureToggles, usageStats usagestats.Service,
+func ProvideService(features featuremgmt.FeatureToggles,
```

Comment:
this import causes a cyclical dependency: 

```
wire: /Users/artur/projects/grafana/grafana/pkg/server/wireexts_oss.go:92:23: cycle for github.com/grafana/grafana/pkg/infra/usagestats.Service:
        github.com/grafana/grafana/pkg/infra/usagestats.Service (github.com/grafana/grafana/pkg/infra/usagestats/service.ProvideService) ->
        github.com/grafana/grafana/pkg/plugins.Store (github.com/grafana/grafana/pkg/plugins/manager.ProvideService) ->
        github.com/grafana/grafana/pkg/plugins.Loader (github.com/grafana/grafana/pkg/plugins/manager/loader.ProvideService) ->
        github.com/grafana/grafana/pkg/plugins.BackendFactoryProvider (github.com/grafana/grafana/pkg/plugins/backendplugin/provider.ProvideService) ->
        *github.com/grafana/grafana/pkg/plugins/backendplugin/coreplugin.Registry (github.com/grafana/grafana/pkg/plugins/backendplugin/coreplugin.ProvideCoreRegistry) ->
        *github.com/grafana/grafana/pkg/tsdb/grafanads.Service (github.com/grafana/grafana/pkg/tsdb/grafanads.ProvideService) ->
        github.com/grafana/grafana/pkg/services/searchV2.SearchService (github.com/grafana/grafana/pkg/services/searchV2.ProvideService) ->
        github.com/grafana/grafana/pkg/services/accesscontrol.AccessControl (github.com/grafana/grafana/pkg/services/accesscontrol/ossaccesscontrol.ProvideService) ->
        github.com/grafana/grafana/pkg/infra/usagestats.Service
wire: github.com/grafana/grafana/pkg/server: generate failed

```


Is it okay if I move this code:
https://github.com/grafana/grafana/blob/1ffbee73ec6e57561889981b69d229e0b649720e/pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol.go#L66-L80

to the `collect` method in `usagestats/statscollector/service.go` to resolve this problem? cc @sakjur

Reply:
Yes, I’d like to move the plugins thing away from usage stats, it’s the most prominent source of these circular dependencies left, but in lieu of that, feel free to move this metric into the collector.

Reply:
as a general solution, what do you think about adding a `ProvidesUsageStats` interface, similar to `CanBeDisabled` / `BackgroundService` ?

```
type ProvidesUsageStats interface {
	GetUsageStats(ctx context.Context) map[string]interface{}
}
```

Reply:
I like that idea!